### PR TITLE
Fix ImmutableConverter and ErrorPath

### LINKED ISSUE #232: Test deserialization of GraphQL Error Path
This library should be able to handle errors like the example in the [spec](https://spec.graphql.org/June2018/#sec-Errors).

The compatibility querstion was raised in https://github.com/graphql-dotnet/graphql-dotnet/pull/1657.

### LINKED ISSUE #234: SystemTextJsonSerializer cannot (de)serialize Nullable<...> types
### Exception
```cs
System.InvalidCastException : Unable to cast object of type 'GraphQL.Client.Serializer.SystemTextJson.ImmutableConverter' to type 'System.Text.Json.Serialization.JsonConverter`1[System.Nullable`1[System.Int32]]'.
   at System.Text.Json.JsonPropertyInfoCommon`4.set_ConverterBase(JsonConverter value)
   at System.Text.Json.JsonPropertyInfo.Initialize(Type parentClassType, Type declaredPropertyType, Type runtimePropertyType, Type implementedPropertyType, PropertyInfo propertyInfo, Type elementType, JsonConverter converter, JsonSerializerOptions options)
   at System.Text.Json.JsonPropertyInfoCommon`4.Initialize(Type parentClassType, Type declaredPropertyType, Type runtimePropertyType, Type implementedPropertyType, PropertyInfo propertyInfo, Type elementType, JsonConverter converter, JsonSerializerOptions options)
   at System.Text.Json.JsonClassInfo.CreateProperty(Type declaredPropertyType, Type runtimePropertyType, Type implementedPropertyType, PropertyInfo propertyInfo, Type parentClassType, JsonConverter converter, JsonSerializerOptions options)
   at System.Text.Json.JsonClassInfo.AddProperty(Type propertyType, PropertyInfo propertyInfo, Type classType, JsonSerializerOptions options)
   at System.Text.Json.JsonClassInfo..ctor(Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializerOptions.GetOrAddClass(Type classType)
   at System.Text.Json.JsonSerializer.GetRuntimeClassInfo(Object value, JsonClassInfo& jsonClassInfo, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.HandleDictionary(JsonClassInfo elementClassInfo, JsonSerializerOptions options, Utf8JsonWriter writer, WriteStack& state)
   at System.Text.Json.JsonSerializer.Write(Utf8JsonWriter writer, Int32 originalWriterDepth, Int32 flushThreshold, JsonSerializerOptions options, WriteStack& state)
   at System.Text.Json.JsonSerializer.WriteCore(Utf8JsonWriter writer, Object value, Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.WriteCore(PooledByteBufferWriter output, Object value, Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.WriteCoreString(Object value, Type type, JsonSerializerOptions options)
   at System.Text.Json.JsonSerializer.Serialize[TValue](TValue value, JsonSerializerOptions options)
   at GraphQL.Client.Serializer.SystemTextJson.SystemTextJsonSerializer.SerializeToString(GraphQLRequest request)
   at GraphQL.Client.Http.GraphQLHttpRequest.ToHttpRequestMessage(GraphQLHttpClientOptions options, IGraphQLJsonSerializer serializer)
   at GraphQL.Client.Http.GraphQLHttpClient.SendHttpRequestAsync[TResponse](GraphQLRequest request, CancellationToken cancellationToken)
   at GraphQL.Client.Http.GraphQLHttpClient.SendQueryAsync[TResponse](GraphQLRequest request, CancellationToken cancellationToken)
...
```

### Reproducing the bug
```cs
        public class WithNullable
        {
            public int? NullableInt { get; set; }
        }

        [Test]
        public void ReproductionImmutableConverterBug()
        {
            var systemTextJsonSerializer = new SystemTextJsonSerializer();
            var serializedString = systemTextJsonSerializer.SerializeToString(new GraphQLRequest
            {
                Query = "{}",
                Variables = new WithNullable
                {
                    NullableInt = 2
                }
            });
        }
```

### PR DESCRIPTION
**Pull request by rose-a**
closes #232 
closes #234

### CODE REVIEW

File: src/GraphQL.Client.Serializer.SystemTextJson/ErrorPathConverter.cs [hunk 1]
```diff
@@ -0,0 +1,48 @@
...
+using System.Text.Json.Serialization;
+
+namespace GraphQL.Client.Serializer.SystemTextJson
+{
+    public class ErrorPathConverter : JsonConverter<ErrorPath>
+    {
+
+        public override ErrorPath Read(ref Utf8JsonReader reader, Type typeToConvert, JsonSerializerOptions options)
+        {
+            using var doc = JsonDocument.ParseValue(ref reader);
```
Comment (sungam3r) [inline src/GraphQL.Client.Serializer.SystemTextJson/ErrorPathConverter.cs:14]: @rose-a Please read this thread https://github.com/graphql-dotnet/server/pull/319

`JsonDocument.ParseValue` is evil

Comment (sungam3r) [reply src/GraphQL.Client.Serializer.SystemTextJson/ErrorPathConverter.cs:14]: I also recommend you to add benchmarks to see what happens. As you can see, there may be unpleasant surprises.

Comment (rose-a) [reply src/GraphQL.Client.Serializer.SystemTextJson/ErrorPathConverter.cs:14]: That's used in the MapConverter, too...

Comment (sungam3r) [reply src/GraphQL.Client.Serializer.SystemTextJson/ErrorPathConverter.cs:14]: The difference in performance is huge.

**Comment by rose-a** [comment]
@sungam3r applied your recommended changes

**Comment by sungam3r** [comment]
Ok. Nevertheless, tests should be added to be sure of improved performance.

**Pull request merged**